File: C/c_linux/DS/linkdlist/L0404/queue.c
#include<stdio.h>
#include<stdlib.h>
#define MAX 5
int rear,front;
int queue[MAX];
void enq(int);
int deq();
int isempty();
int isfull();
void display();
main()
{
	int ch,data;
	rear=front=-1;
	while(1)
	{
		printf("1.enq\n2.deq\n3.display\n4.exit\n");
		printf("enter ch:");
		scanf("%d",&ch);
		switch(ch)
		{
			case 1:printf("enter data:");
				scanf("%d",&data);
				if(isfull())
					printf("queue is full\n");
				else
					enq(data);
				break;
			case 2:if(isempty())
				printf("queue is empty\n");
				else
				{	data=deq();
					printf("%d",data);
				}
				break;
			case 3:display();break;
			case 4:exit(0);
			default:printf("invalid ch\n");break;	

		}
	}

}
void enq(int data)
{
	if(front==-1)
		front++;
	rear++;
	queue[rear]=data;
}
int deq()
{
	int item;
	item=queue[front];
	front++;
	if(front==MAX)
		rear=front=-1;
	return item;

}
int isempty()
{
	if(front==-1)
		return 1;
	else
		return 0;
}
int isfull()
{
	if(rear==MAX-1)
		return 1;
	else
		return 0;

}
void display()
{
	int i=front;
	while(i<=rear)
	{
		printf("%d",queue[i]);
		i++;	

	}

}

File: python/assign/modul/queue.py
#!/usr/bin/python
queue_max=5
rear=front=-1
list1=[]
def isfull():
	if rear==queue_max-1:
		return True
	else:
		return False
def isempty():
	if front==-1:
		return True
	else:
		return False
print "1.enq\n2.deq\n3.display\n4.exit"
choice=input("enter choice:")
while choice!=4:
	if choice==1:
		if isfull():
			print "queue is full"
		else:
			if(front==-1):
				front+=1
			rear+=1
			num = input("enter val into queue:")
			list1.append(num)
	elif choice==2:
		if isempty():
			print "queue is empty"
		else:
			del(list1[front])
			rear-=1
#			front+=1
			if front-1==rear:
				front=rear=-1
	elif choice==3:
		print list1
	print "1.enq\n2.deq\n3.display\n4.exit"
	choice=input("enter choice:")


